ankura/pipeline.py

```diff
@@ -614,15 +615,44 @@ def build_docwords(corpus, V=None):
 
 
 def test_train_split(corpus, num_train=None, num_test=None, **kwargs):
-    if not num_train:
+    if not num_train and not num_test:
         num_train = int(len(corpus.documents) * .8)
-    if not num_test:
+        num_test = len(corpus) - num_train
+    elif not num_train:
+        num_train = len(corpus.documents) - num_test
+    elif not num_test:
         num_test = len(corpus.documents) - num_train
 
-    doc_ids = np.random.permutation(len(corpus.documents))
-    train_ids, test_ids = doc_ids[:num_train], doc_ids[num_train: num_train+num_test]
-    train = Corpus([corpus.documents[d] for d in train_ids], corpus.vocabulary, corpus.metadata)
-    test = Corpus([corpus.documents[d] for d in test_ids], corpus.vocabulary, corpus.metadata)
+    print("Splitting test and train")
+
+    try:
+        
+        doc_ids = np.random.permutation(len(corpus.documents))
+        train_ids, test_ids = doc_ids[:num_train], doc_ids[num_train: num_train+num_test]
+        train = Corpus([corpus.documents[d] for d in train_ids], corpus.vocabulary, corpus.metadata)
+        test = Corpus([corpus.documents[d] for d in test_ids], corpus.vocabulary, corpus.metadata)
+
+    #This occurs when your corpus is a doc stream instead of a list of documents.
```

Comment:
More specifically, when the documents doesn't support random indexing. Be precise please. ![bitmoji](https://render.bitstrips.com/v2/cpanel/27b025e9-96e0-4fb8-af6d-50d0d1281753-58e1cf7a-21f2-4c43-82cb-72edb2944d01-v1.png?transparent=1&palette=1&width=246)

Reply:
![bitmoji](https://render.bitstrips.com/v2/cpanel/3c97455d-0ce1-4ac8-82a0-601e6d6f6fd2-b30ef521-856b-4209-82ef-10378d6686fd-v1.png?transparent=1&palette=1&width=246)